### PR TITLE
storybook: Publish to a mirror repo

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

The "Storybook GitHub Pages" and "Build" actions were both building,
with the latter's result being thrown away. We've decided to drop the
former and have the "Build" action publish it to a mirror repo like we
do for all the other projects.

#### Jetpack product discussion

p1629484301244500-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Check the jetpack-build artifact produced by the "Build" workflow. See that Automattic/jetpack-storybook is present in the archive, with the appropriate contents.
* Merge this, wait for the publish, configure Pages in the mirror repo, and see that the content shows properly at https://automattic.github.io/jetpack-storybook/.

### CODE REVIEW

File: projects/js-packages/storybook/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
 		"validate-es5": "eslint --parser-options=ecmaVersion:5 --no-eslintrc --no-ignore",
 		"storybook:build": "build-storybook -c ./storybook -o ./storybook/build",
 		"dev:packages": "echo 'TODO dev:packages'",
-		"storybook:dev": "concurrently \"npm run dev:packages\" \"start-storybook -c ./storybook -p 50240\""
+		"storybook:dev": "concurrently \"pnpm run dev:packages\" \"start-storybook -c ./storybook -p 50240\""
```
Comment (gravityrail) [inline projects/js-packages/storybook/package.json:27]: Good catch!

**Review by gravityrail — CHANGES_REQUESTED** [review]
How is this triggering the `storybook:build` step now that it's being published to the mirror? It's not run by default when you run `pnpm run build` or anything like that.

**Comment by anomiex** [comment]
Same as it was before: the "Build" workflow does `jetpack build js-packages/storybook -v --production`. See https://github.com/Automattic/jetpack/runs/3385254753?check_suite_focus=true#step:11:3539 for example.

The difference is that now it'll go on to push the result to the mirror repo at https://github.com/Automattic/jetpack-storybook (when it's run in response to a merge to master).

**Comment by gravityrail** [comment]
Oh, I see now, it's triggered by `composer run build-development` in the composer.json file. It wasn't part of this diff so I forgot it was there. Approving!

**Review by gravityrail — APPROVED** [review]

**Review by kraftbj — APPROVED** [review]

**Pull request merged**